pkg/controller/agent/nad/controller.go

```diff
@@ -70,13 +67,17 @@ func (h Handler) OnChange(key string, nad *nadv1.NetworkAttachmentDefinition) (*
 	}
 
 	klog.Infof("nad configuration %s has been changed: %s", nad.Name, nad.Spec.Config)
-	netconf := &hn.NetConf{}
-	if err := json.Unmarshal([]byte(nad.Spec.Config), netconf); err != nil {
-		return nil, err
-	}
-
 	// TODO delete previous vlan id when update nad
 
+	vlanStr, ok := nad.Labels[utils.KeyVlanLabel]
+	if !ok {
+		return nad, nil
+	}
+	vlanID, err := strconv.Atoi(vlanStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid vlan %s", vlanStr)
+	}
```

Comment:
seems a duplicate code in many places, can we refactor it?

Reply:
Both the VLAN id with string format and integer format are required. I think it's fine here.